subiquity/models/subiquity.py

```diff
@@ -28,3 +31,81 @@ def __init__(self, common):
         self.network = NetworkModel()
         self.filesystem = FilesystemModel(common['prober'])
         self.identity = IdentityModel()
+
+    def _cloud_init_config(self):
+        user = self.identity.user
+        users_and_groups_path = os.path.join(os.environ.get("SNAP", "/does-not-exist"), "users-and-groups")
+        if os.path.exists(users_and_groups_path):
+            groups = open(users_and_groups_path).read().split()
+        else:
+            groups = ['admin']
+        groups.append('sudo')
+        user_info = {
+            'name': user.username,
+            'gecos': user.realname,
+            'passwd': user.password,
+            'shell': '/bin/bash',
+            'groups': groups,
+            'lock-passwd': False,
+            }
+        if user.ssh_import_id is not None:
+            user_info['ssh_import_id'] = [user.ssh_import_id]
+        # XXX this should set up the locale too.
+        return {
+            'users': [user_info],
+            'hostname': self.identity.hostname,
+        }
+
+    def _write_files_config(self):
+        # TODO, this should be moved to the in-target cloud-config seed so on first
+        # boot of the target, it reconfigures datasource_list to none for subsequent
+        # boots.
+        # (mwhudson does not entirely know what the above means!)
```

Comment:
It means, we should not re-initialize a server instance, if one wipes `/var/lib/cloud/instance-id` state and like reset user changed passwords. Or e.g. randomly detect a usb stick with a config drive, and randomly reinitialise this server with that usb stick data.

On the other hand, storing this enables one to easily "reset" this server back to original passwords....

Reply:
OK, that makes sense I guess. How do we do that? Add a cloud-init runcmd that runs something like rm -rf /var/lib/cloud/seed/nocloud-net or echoes "datasource_list: None" into some file in etc/cloud? (should we anyway set datasource_list to [NoCloud, None]?)